core/primitives/src/crypto.rs

```diff
@@ -625,6 +625,31 @@ pub trait TypedKey {
 	const KEY_TYPE: KeyTypeId;
 }
 
+/// An identifier for a key provider.
+///
+/// 0-1024 are reserved.
+pub type KeyProviderId = u32;
+
+/// Constant key proviers.
+pub mod key_providers {
+	use super::KeyProviderId;
+
+	/// Aura key provider.
+	pub const AURA: KeyProviderId = 10;
```

Comment:
Er... no.

core shouldn't be hard-coding consensus-algorithm-related stuff in here.